### ISSUE TITLE
Take up a lot of disk space

### ISSUE THREAD

**Issue by WClouds** [rwynn/monstache#148]
I was using a docker built 5 days ago. I have 2 collections to sync. I run monstache on an AWS EC2 instance. The disk of the instance is 8GB and memory is 4GB. After I start monstache, it quickly used up all my disk space.
Is it normal? How much disk it takes?
Below is my monstache.config.yml
```
namespace-regex = '^live.(orders|events)$'
direct-read-namespaces = ["live.events", "live.orders"]
change-stream-namespaces = ["live.events", "live.orders"]
gzip = true
stats = false
index-stats = false
verbose = true
exit-after-direct-reads = false
pprof = true
enable-http-server = true
```

**Comment by rwynn** [comment]
It doesn’t write to disk by default. Are you redirecting standard out to a file?  

Also verbose is meant only for trouble shooting. I outputs the whole bulk request response to standard out by default.

**Comment by rwynn** [comment]
If you mean the Elasticsearch disk is filling up then that just means your MongoDB data is too large to hold it. 

What are the sizes of those 2 collections?

**Comment by WClouds** [comment]
I found that the problem was caused by the docker's log. I can give a limit to solve it.
In addition, I looked at the size of my collection, and their data is very large. Can I choose to only sync data after a certain point in time?

**Comment by rwynn** [comment]
You can create a view in MongoDB such that the view is a subset of the entire collection.  

https://docs.mongodb.com/manual/core/views/

The view can be created with a pipeline.

https://docs.mongodb.com/manual/core/aggregation-pipeline/

The pipeline might be such that it uses a `$match` stage.  If you have a date field in your data your could limit the view to a subset of the data where this date is greater than some date in the past. 

Once you've created the view, then use the namespace of the view in your direct-reads instead of the collection.  E.g. `live.events-view`.

An alternative to creating a view in MongoDB is to setup a pipeline query in Monstache.  

https://rwynn.github.io/monstache-site/advanced/#aggregation-pipelines

**Comment by WClouds** [comment]
I found I can set a `Filtering ` for Monstache.
So can I just filter the timestamp?

**Comment by WClouds** [comment]
Yes, I am the same as you think.
I think I can test it.

**Comment by WClouds** [comment]
The `Filter` works. Thank you!
And one more question, if I set direct-read-namespaces and change-stream-namespaces, does it mean every time I run Monstache, it will directly copy the collections from MongoDB to Elasticsearch. And If I don't set direct-read-namespaces and change-stream-namespaces, it will just copy data by listening event?
What I want to know is, if I set direct-read-namespaces and change-stream-namespaces, and everytime I restart Monstache, does it copy the collections again and again?
